vm/sdk/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use athena_hostfunctions;
+use athena_interface::{Address, Balance, Bytes32, BYTES32_LENGTH};
+use athena_vm::helpers::{address_to_32bit_words, balance_to_32bit_words};
+
+pub type Pubkey = Bytes32;
+pub const PUBKEY_LENGTH: usize = BYTES32_LENGTH;
+
+pub fn call(address: Address, input: Option<Vec<u8>>, amount: Balance) {
+  let address = address_to_32bit_words(address);
+  let amount = balance_to_32bit_words(amount);
+
+  // for now, require input to be word-aligned
+  // we can pad the input but need to know more about the contents
+  let (input, input_len) = if let Some(input) = input {
+    if (input.len() % 4) != 0 {
+      panic!("input is not byte-aligned");
+    }
+    (
+      input
+        .chunks(4)
+        .map(|chunk| {
+          let mut bytes = [0u8; 4];
+          bytes.copy_from_slice(chunk);
+          u32::from_le_bytes(bytes)
+        })
```

Comment:
As nightly is already required anyway, you could use its nighly-only features like `array_chunks`:
```suggestion
      input
        .array_chunks()
        .map(|chunk| { u32::from_le_bytes(*chunk) })
```

Note: you would need to add `#![feature(array_chunks)]` at the top of the file to enable it.

Reply:
Btw, why is nightly used? The code and tests compile on the latest stable.